extensions/wikia/PowerUser/PowerUser.class.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Class to manage data on wikia's PowerUsers.
+ * It implements methods to add and remove
+ * poweruser properties and log these actions
+ * in a consistent way.
+ *
+ * @package PowerUser
+ * @author Adam Karmiński <adamk@wikia-inc.com>
+ */
+
+namespace Wikia\PowerUser;
+
+use Wikia\Logger\WikiaLogger;
+
+class PowerUser {
+	/**
+	 * Names of properties used to described PowerUsers
+	 */
+	const TYPE_ADMIN = 'poweruser_admin';
+	const TYPE_FREQUENT = 'poweruser_frequent';
+	const TYPE_LIFETIME = 'poweruser_lifetime';
+
+	/**
+	 * Requirement to meet to become a PowerUser
+	 */
+	const MIN_LIFETIME_EDITS = 2000;
+	const MIN_FREQUENT_EDITS = 140;
+
+	const LOG_MESSAGE = 'PowerUsersLog';
+	const ACTION_ADD = 'add';
+	const ACTION_REMOVE = 'remove';
+
+	/**
+	 * A table of all poweruser properties' names
+	 * for in_array checks
+	 * @var array
+	 */
+	public static $aPowerUserProperties = [
+		self::TYPE_ADMIN,
+		self::TYPE_FREQUENT,
+		self::TYPE_LIFETIME,
+	];
+
+	/**
+	 * An array of the names of groups defining
+	 * PowerUsers of an admin type
+	 * @var array
+	 */
+	public static $aPowerUserAdminGroups = [
+		'sysop',
+	];
+
+	private $oUser;
+
+	function __construct( \User $oUser ) {
+		$this->oUser = $oUser;
+	}
+
+	/**
+	 * Adds a given PowerUser property to a user
+	 *
+	 * @param $sProperty string One of the types in consts
```

Comment:
The right order for the docs is [`@param [type] [varname]`](https://www.mediawiki.org/wiki/Manual:Coding_conventions/PHP#Comments_and_documentation), so this should be `@param string $sProperty` (same in a few of the other method docs).